### PR TITLE
 LF can Create an engineers’ list belonging to them.

### PR DESCRIPTION
**Pull request by Bahatiroben**
#### What does this PR do?
it allows LF and super LF to add engineers to their list and removing them from their list
#### Description of Task to be completed?
- as an LF, I can add/remove an engineer or a list of engineers to my the list of my engineers
- engineers need to be registered to dev-pulse
- LF needs to be logged in

#### How should this be manually tested?
- clone the repo
- go to branch `ft-lf-add-engineers-list-169153727`
- login as an LF
PATCH `api/v1/group`
payload: ARRAY OF IDs
```
{
"engineers": [1,2,3]
}
```
#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
[#169153727](https://www.pivotaltracker.com/story/show/169153727)
#### Screenshots (if appropriate)
N/A
#### Questions:
N/A

### CODE REVIEW

File: src/controllers/groupController.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+import engineerService from '../services/engineerService';
+import userService from '../services/userService';
+import Response from '../helpers/response';
+
+class EngineerService {
+  async createOrRemove(req, res, next) {
+    try {
+      const { engineers } = req.body;
+      const { user } = req;
+      if (user.role !== 'LF' && user.role !== 'super LF') return Response.authorizationError(res, 'you have no rights to perform this operation');
```
Comment (willywax) [inline src/controllers/groupController.js]: we can make the check shorter if we check if user role is an engineer instead

File: src/services/userService.js [hunk 2]
```diff
@@ -37,6 +37,22 @@ class UserService {
...
+    try {
+      const users = await User.findOne({
+        where: param,
+      });
+      return users;
+    } catch (error) {
+      throw error;
+    }
+  }
+
```
Comment (willywax) [inline src/services/userService.js]: We already have a method for Finding one user in user services

File: src/controllers/groupController.js (hunk 1 above)
Comment (willywax) [inline src/controllers/groupController.js]: we can make this short by checking if user.role is an Engineer instead

**Review by willywax — COMMENTED** [review]
Looks great. Left a little feedback on groupController.js

**Review by willywax — APPROVED** [review]
Looks good.

**Review by drayzii — APPROVED** [review]
LGTM

**Pull request merged**